test/api/stack.test.ts

```diff
@@ -0,0 +1,64 @@
+import "@aws-cdk/assert/jest";
+import * as cdk from "@aws-cdk/core";
+import * as fs from "fs";
+import { ApiStack } from "../../lib/api/stack";
+import { PoolManagementStack } from "../../lib/pool-management/stack";
+import { TenantManagementStack } from "../../lib/tenant-management/stack";
+import * as path from "path";
+
+let poolManagementStack: PoolManagementStack;
+let tenantManagementStack: TenantManagementStack;
+let app: cdk.App;
+
+beforeEach(() => {
+  app = new cdk.App({
+    context: {
+      tenants: [],
+    },
+  });
+
+  poolManagementStack = new PoolManagementStack(app, "pool-management", { deployStage: "test" });
+  tenantManagementStack = new TenantManagementStack(app, "tenant-management", {
+    poolTable: poolManagementStack.poolTable,
+    deployStage: "test",
+  });
+});
+
+test("GraphQL APi with name kiko-api is created", () => {
+  const stack = new ApiStack(app, "api", {
+    userPool: tenantManagementStack.userPool,
+    poolTable: poolManagementStack.poolTable,
+    recipientTable: poolManagementStack.recipientTable,
+    testResultProcessingStateMachine: poolManagementStack.testResultProcessingStateMachine,
+  });
+  expect(stack).toHaveResource("AWS::AppSync::GraphQLApi", {
+    Name: "kiko-api",
+  });
+});
+
+test("GraphQL Schema matches source schema definition", async (done) => {
```

Comment:
No need to `async` if you don't `await`. You could also get rid of the callback parameter altogether if you use `fs.readFileSync` down below.

Reply:
With regards to the DynamoDb stream: exactly. This was the idea. 
The initial attempt was to use a Lambda resolver and simply list the endpoints from Pinpoint. But Pinpoint itself does not support listing all endpoints at all. You can only export them to S3.